### PR TITLE
ci: run sqllogic test with docker image

### PR DESCRIPTION
**Pull request by everpcpc**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Fixes #issue

### CODE REVIEW

**Comment by TCeason** [comment]
The ci err is strange. Cannot be reproduced locally.

**Review by TCeason — APPROVED** [review]

**Comment by TCeason** [comment]
Runner: mysql
[1146](https://github.com/datafuselabs/databend/runs/8291130844?check_suite_focus=true#step:4:1152)
ErrorType: statement query get result not equal to expected
[1147](https://github.com/datafuselabs/databend/runs/8291130844?check_suite_focus=true#step:4:1153)
Message: 
[1148](https://github.com/datafuselabs/databend/runs/8291130844?check_suite_focus=true#step:4:1154)
 Expected:
[1149](https://github.com/datafuselabs/databend/runs/8291130844?check_suite_focus=true#step:4:1155)
2021-08-30 10:30:00                                                             
[1150](https://github.com/datafuselabs/databend/runs/8291130844?check_suite_focus=true#step:4:1156)
 Actual:
[1151](https://github.com/datafuselabs/databend/runs/8291130844?check_suite_focus=true#step:4:1157)
2021-08-30 10:30:00.000000                                                      
[1152](https://github.com/datafuselabs/databend/runs/8291130844?check_suite_focus=true#step:4:1158)
 Statement:
[1153](https://github.com/datafuselabs/databend/runs/8291130844?check_suite_focus=true#step:4:1159)
Parsed Statement
[1154](https://github.com/datafuselabs/databend/runs/8291130844?check_suite_focus=true#step:4:1160)
    at_line: 259,
[1155](https://github.com/datafuselabs/databend/runs/8291130844?check_suite_focus=true#step:4:1161)
    s_type: Statement: query, type: T, query_type: T, retry: False,
[1156](https://github.com/datafuselabs/databend/runs/8291130844?check_suite_focus=true#step:4:1162)
    suite_name: base/02_function/02_0012_function_datetimes,

**Comment by TCeason** [comment]
Runner: mysql
ErrorType: statement query get result not equal to expected
Message: 
 Expected:
2021-04-30 22:48:31                                                             
 Actual:
2021-04-30 22:48:31.000000                                                      
 Statement:
Parsed Statement
    at_line: 184,
    s_type: Statement: query, type: T, query_type: T, retry: False,
    suite_name: base/02_function/02_0012_function_datetimes_tz,
    text:
        select to_start_of_second(to_timestamp(1619822911999000));
    results: [(<re.Match object; span=(0, 4), match='----'>, 188, '2021-04-30 22:48:31')],
    runs_on: {'http', 'clickhouse', 'mysql'},
 Start Line: 188, Result Label:

**Comment by Xuanwo** [comment]
Maybe this problem is related to timezone? Have to set timezone correctly in the docker runner?

File: tests/logictest/mysql_runner.py [hunk 1]
```diff
@@ -93,5 +93,8 @@ def execute_query(self, statement):
                     log.debug(
                         f"Unknown type {query_type[i]} in query {statement.text} row {ri} col {i} value {v}"
                     )
-                vals.append(str(v))
+                if isinstance(v, datetime):
+                    vals.append(v.strftime('%Y-%m-%d %H:%M:%S.%f'))
```
Comment (TCeason) [inline tests/logictest/mysql_runner.py:97]: Maybe this can not cover all solutions.

Like function to_start_of_second

**Comment by TCeason** [comment]
> Maybe this problem is related to timezone? Have to set timezone correctly in the docker runner?

No. The first err like this:

https://github.com/datafuselabs/databend/runs/8285682330?check_suite_focus=true

write a demo:

```
➜  source-code cat a.py 
import mysql.connector
from mysql.connector import Error
try:
    connection = mysql.connector.connect(host='[127.0.0.1](http://127.0.0.1/)',
                                         database='default',
                                         user='root',
                                         password='',port='3307')
    if connection.is_connected():
        db_Info = connection.get_server_info()
        print("Connected to MySQL Server version ", db_Info)
        cursor = connection.cursor()
        cursor.execute("select to_datetime(1630320462000000)")
        r = cursor.fetchall()
        print(r)
        for (ri, row) in enumerate(r):
            print("ri is: ", ri, " row is: ", row)
            for (i, v) in enumerate(row):
                print("i is: ", i, "v is: ", v)

except Error as e:
    print("Error while connecting to MySQL", e)
finally:
    if connection.is_connected():
        cursor.close()
        connection.close()
        print("MySQL connection is closed")
➜  source-code python a.py
Connected to MySQL Server version  8.0.26-v0.8.16-nightly-c73f3fb-simd(rust-1.64.0-nightly-2022-09-11T03:23:17.763220997Z)
[(datetime.datetime(2021, 8, 30, 10, 47, 42),)]
ri is:  0  row is:  (datetime.datetime(2021, 8, 30, 10, 47, 42),)
i is:  0 v is:  2021-08-30 10:47:42
MySQL connection is closed
```

The result does not have ms.

**Comment by TCeason** [comment]
> Runner: mysql ErrorType: statement query get result not equal to expected Message: Expected: 2021-04-30 22:48:31 Actual: 2021-04-30 22:48:31.000000 Statement: Parsed Statement at_line: 184, s_type: Statement: query, type: T, query_type: T, retry: False, suite_name: base/02_function/02_0012_function_datetimes_tz, text: select to_start_of_second(to_timestamp(1619822911999000)); results: [(<re.Match object; span=(0, 4), match='----'>, 188, '2021-04-30 22:48:31')], runs_on: {'http', 'clickhouse', 'mysql'}, Start Line: 188, Result Label:

Maybe we can cast the return type in test.

Like this:

select to_start_of_second(to_timestamp(1619822911999000))::String;

**Comment by TCeason** [comment]
The test fail reason is the mysql connector is 8.0.30 in docker image.

Need degre to 8028 and fix the err in 8030

**Pull request merged**